frontend/amundsen_application/static/js/ducks/tableMetadata/api/helpers.ts

```diff
@@ -96,3 +100,15 @@ export function createOwnerNotificationData(
 export function shouldSendNotification(user: PeopleUser): boolean {
   return user.is_active && !!user.display_name;
 }
+
+
+/**
+ * Returns total column count, including nested columns
+ */
+export function getColumnCount(columns: TableColumn[]) {
+  let count = columns.length;
+  columns.forEach((column) => {
+    count += column?.children?.length || 0;
+  });
+  return count;
```

Comment:
Another option, without using forEach and without mutating  the 'count' variable.

```suggestion
return columns.reduce((acc, column) => {
  return acc + column?.children?.length || 0;
}, columns.length);
```